book/book_crud/views.py

```diff
@@ -111,14 +112,24 @@ def user_list(request):
     serializer = UserSerializer(books, many=True)
     return Response(serializer.data)
 
-@api_view(["POST"])
-def create_library(request):
-    if request.method == "POST":
-        serializer = LibrarySerializer(data=request.data)
-        if serializer.is_valid():
-            serializer.save()
-            return Response(serializer.data, status=HTTP_201_CREATED)
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+# @api_view(["POST"])
+# def create_library(request):
+#     if request.method == "POST":
+#         serializer = LibrarySerializer(data=request.data)
+#         if serializer.is_valid():
+#             serializer.save()
+#             return Response(serializer.data, status=HTTP_201_CREATED)
+#         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class LibraryView(generics.CreateAPIView):
+    serializer_class = LibrarySerializer
+
+    def get_serializer(self,*args,**kwargs):
+        if isinstance(kwargs.get("data",{}),list):
+            kwargs["many"] = True
+        return super(LibraryView,self).get_serializer(*args,**kwargs)
+
```

Comment:
class RegisterView(CreateAPIView):
    permission_classes = [permissions.IsAuthenticated]
    serializer_class = UserSerializer
    you can also try this.

Reply:
not a register view this is a library create view